torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3150,7 +3150,7 @@ def test_ddp_logging_data(self):
             self.assertEqual(ddp_logging_data.rank, dist.get_rank())
             self.assertEqual(ddp_logging_data.backend, 'gloo')
             self.assertEqual(ddp_logging_data.module_name, 'Net')
-            self.assertEqual(ddp_logging_data.device_ids, [])
+            self.assertEqual(ddp_logging_data.device_ids, "")
```

Comment:
Is this empty because DDP is on a CPU device? Could we test it with GPU device and ensure that they are appropriately logged? 

Also, if it is CPU device looks like below output_device is -1 which is also CPU device. Can we make this consistent?

Reply:
you mean make "output_device" as string, and output it as "" when they are none. That sounds good to me if they result in confusion.